blockchain/blockchain.go

```diff
@@ -1405,9 +1401,6 @@ func (bc *BlockChain) writeBlockWithStateParallel(block *types.Block, receipts [
 			"hash", block.Hash(), "parentHash", block.ParentHash())
 		return WriteResult{Status: NonStatTy}, consensus.ErrUnknownAncestor
 	}
-	// Make sure no inconsistent state is leaked during insertion
-	bc.mu.Lock()
-	defer bc.mu.Unlock()
```

Comment:
@hqjang-pepper Removing this is okay?

Reply:
There are only 2 cases where `WriteBlockWithState` is used.
1. `wait()` in `worker.go` -> protects worker's state with `stateMu`.
2. `InsertChain()` in `blockchain.go` -> protects state with `mu`.

I think `stateMu` cannot protect state in chain.
Additional version of `writeBlockwithState` (which contains `mu` lock/unlock, for sake of `worker.go`)will be needed.

thanks for your advice!